client/a8c-for-agencies/components/a4a-feedback/hooks/use-provide-general-feedback.ts

```diff
@@ -0,0 +1,181 @@
+import type { SubmitContactSupportParams } from 'calypso/a8c-for-agencies/data/support/types';
+
+export const useProvideGeneralFeedback = () => {
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+
+	const agencyId = useSelector( getActiveAgencyId );
+	const user = useSelector( getCurrentUser );
+
+	const [ createTicket, setCreateTicket ] = useState( false );
+
+	const getMessage = useCallback( ( type: FeedbackType ) => {
+		switch ( type ) {
+			case FeedbackType.BugReport:
+				return 'A4A Feedback - Bug Report';
+			case FeedbackType.SuggestAFeature:
+				return 'A4A Feedback - Suggest a Feature';
+		}
+	}, [] );
+
+	const {
+		mutate: saveFeedback,
+		isPending: isSavingFeedback,
+		isSuccess: isSuccessFeedback,
+		isError: isErrorFeedback,
+	} = useSaveFeedbackMutation();
+
+	const {
+		mutateAsync: submitSupportForm,
+		isPending: isSubmittingSupportForm,
+		isSuccess: isSuccessSupportForm,
+		isError: isErrorSupportForm,
+	} = useSubmitSupportFormMutation();
+
+	const isSuccess = useMemo(
+		() => ( createTicket ? isSuccessSupportForm && isSuccessFeedback : isSuccessFeedback ),
+		[ createTicket, isSuccessSupportForm, isSuccessFeedback ]
+	);
+
+	const isError = useMemo(
+		() => ( createTicket ? isErrorSupportForm || isErrorFeedback : isErrorFeedback ),
+		[ createTicket, isErrorSupportForm, isErrorFeedback ]
+	);
+
+	const handleError = useCallback( () => {
+		dispatch(
+			errorNotice(
+				translate( 'An error occurred while submitting your feedback. Please try again.' ),
+				{
+					id: 'a4a-feedback-error',
+					duration: 10000,
+				}
+			)
+		);
+		setCreateTicket( false );
+	}, [ dispatch, translate ] );
+
+	const handleSuccess = useCallback( () => {
+		dispatch(
+			successNotice(
+				translate(
+					'Thanks! Our team will use your feedback to help prioritize improvements to Automattic for Agencies.'
+				),
+				{
+					id: 'a4a-feedback-success',
+					duration: 10000,
+				}
+			)
+		);
+		setCreateTicket( false );
+	}, [ dispatch, translate ] );
+
+	useEffect( () => {
+		if ( isError ) {
+			handleError();
+		}
+		if ( isSuccess ) {
+			handleSuccess();
+		}
+	}, [ isError, isSuccess, handleError, handleSuccess, createTicket ] );
+
+	const handleSaveFeedback = useCallback(
+		( feedback: GeneralFeedbackParams ) => {
+			if ( ! agencyId || ! feedback.responses ) {
+				return;
+			}
+			const { responses, type, ticketId } = feedback;
+			// Convert the responses object to a format that can be sent so that we can send the correct data to MC Marketing Survey tool
+			const surveyResponses = Object.entries( responses ).reduce(
+				( acc, [ key, value ] ) => ( {
+					...acc,
+					[ key ]: { text: value },
+				} ),
+				{}
+			);
+			const params: FeedbackSurveyResponsesPayload = {
+				site_id: agencyId,
+				survey_id: type,
+				survey_responses: { ...surveyResponses, ...( ticketId ? { ticket_id: ticketId } : {} ) },
+			};
+			saveFeedback( { params } );
```

Comment:
Can we just call `handleSuccess` here? Ultimately, regardless of whether we create a ticket or not, we always end up having this operation as the last call. Therefore, we only care about this being a success instead of creating the isSuccess memo and handling it on useEffect?

```

			saveFeedback(
				{ params },
				{
					onSuccess: () => {
						handleSuccess();
					},
				}
			);
```

Reply:
Great call-out! I made some changes later on, and missed updating this.